stack_image.go

```diff
@@ -0,0 +1,174 @@
+package hsup
```

Comment:
This is interesting functionality. Can you write a top-level comment on this file describing what's going on and what purpose it serves? I realize that the status quo for this project is below that threshold of software engineering process, but as we're starting to bloat the number of contributors these tweaks are going to become more useful.